src/Interpreters/TableJoin.cpp

```diff
@@ -474,4 +462,52 @@ String TableJoin::renamedRightColumnName(const String & name) const
     return name;
 }
 
+void TableJoin::addJoinCondition(const ASTPtr & ast, bool is_left)
+{
+    LOG_TRACE(&Poco::Logger::get("TableJoin"), "Add join condition for {} table: {}", (is_left ? "left" : "right"), queryToString(ast));
+
+    if (is_left)
+        on_filter_condition_asts_left.push_back(ast);
+    else
+        on_filter_condition_asts_right.push_back(ast);
+}
+
+ASTPtr TableJoin::buildJoinConditionColumn(const ASTs & on_filter_condition_asts)
```

Comment:
What is an expected resulting AST? I saw that we can push `ASTIdentifier` along with `ASTFunction` into `on_filter_condition_asts`.

Reply:
It's expression containing `ASTIdentifier` or `ASTFunction` from `ON` section for one table joined with `AND`. I'll add a comment

Reply:
`buildJoinConditionColumn` is private static function, I added comment to TableJoin.h before related functions (addJoinCondition, joinConditionColumn) with general description
https://github.com/vdimir/ClickHouse/blob/de3198020cfa05275c9c63a16657fd2f18ec3d45/src/Interpreters/TableJoin.h#L162-L177